### PR TITLE
18717: Fixes issue where null counts were not always updated during train, edit, or remove

### PR DESCRIPTION
**Pull request by howsoRes**

### CODE REVIEW

File: trainee_template/train.amlg [hunk 1]
```diff
@@ -397,6 +397,23 @@
...
 				)
 
+				;update count of nulls for the feature after training
+				(accum_to_entities (assoc
+					featureNullRatiosMap
+						(map
+							(lambda (let
+								(assoc feature (current_index 1))
+								(set
+									;if it hasn't been computed yet, create an assoc of num_nulls -> number of nuls for the feature
```
Comment (calebwherry) [inline trainee_template/train.amlg]: *nulls (last one)

File: trainee_template/train.amlg [hunk 2]
```diff
@@ -397,6 +397,23 @@
 					)
 				)
 
+				;update count of nulls for the feature after training
```
Comment (howsohazard) [inline trainee_template/train.amlg]: Break into separate method (train is already too long).  Make sure that case editing and removing updates this correctly as well; may be able to reuse the method.

**Review by howsohazard — CHANGES_REQUESTED** [review]

File: trainee_template/marginal.amlg [hunk 3]
```diff
@@ -499,6 +499,26 @@
...
+		(accum_to_entities (assoc
+			featureNullRatiosMap
+				(map
+					(lambda (let
+						(assoc feature (current_index 1))
+						(set
+							;if it hasn't been computed yet, create an assoc of num_nulls -> number of nulls for the feature
+							(or (get featureNullRatiosMap feature) (assoc))
+							"num_nulls"
+							(size (contained_entities (list (query_equals feature (null)))))
```
Comment (howsohazard) [inline trainee_template/marginal.amlg]: This should be a compute_on_contained_entities with a query_count, which is far more efficient in memory and performance.

Comment (howsohazard) [reply trainee_template/marginal.amlg]: I found at least one other (size (contained_entities... in analyzing boundaries for a feature.

**Review by howsohazard — APPROVED** [review]

**Pull request merged**